engine/flink/model-util/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/AggregatorFunction.scala

```diff
@@ -98,11 +98,15 @@ trait AggregatorFunctionMixin[MapT[K,V]] { self: StateHolder[MapT[Long, AnyRef]]
                                   out: Collector[ValueWithContext[AnyRef]]): MapT[Long, aggregator.Aggregate] = {
     val newElementInStateTimestamp = computeTimestampToStore(timestamp)
     val newElement = value.value.value.asInstanceOf[aggregator.Element]
-    val newState = computeNewState(newElementInStateTimestamp, newElement)
+    val (newState, stateWasChanged) = computeNewState(newElementInStateTimestamp, newElement)
 
-    updateState(newState.asInstanceOf[MapT[Long, AnyRef]], newElementInStateTimestamp + timeWindowLengthMillis, timeService)
+    if (stateWasChanged) {
+      updateState(newState.asInstanceOf[MapT[Long, AnyRef]], newElementInStateTimestamp + timeWindowLengthMillis, timeService)
+      retrievedBucketsHistogram.update(newState.toScalaMapRO.size)
```

Comment:
I think retrievedBucketsHistogram name suggests we count retrieved, not updated? Maybe we can add separate metric for that?

Reply:
Ahh I forgot about comment how it works. Let's keep only `retrievedBucketsHistogram` - not sure if `updatedBucketsHistogram` is really needed